sysbox-in-docker/Makefile

```diff
@@ -21,12 +17,29 @@ SIND_IMAGE := nestybox/sysbox-in-docker
 	ubuntu-bionic   \
 	ubuntu-focal
 
+# Host kernel info
+KERNEL_REL := $(shell uname -r)
+export KERNEL_REL
+
+# Sysbox image-generation globals utilized during the sysbox's building and testing process.
+IMAGE_BASE_DISTRO := $(shell lsb_release -is | tr '[:upper:]' '[:lower:]')
```

Comment:
These globals are already defined in sysbox/Makefile. Why do we need to redefine them here?

Reply:
We need to redefine them because we don't want to depend from the Sysbox Makefile for the sysbox-in-docker Makefile. By breaking this dependency, we can now invoke the sysbox-in-docker make target from within the new sysbox-in-docker test container easily.

Reply:
In general, Makefiles in sub-dirs should not depend on Makefiles in parent dirs.

Reply:
> We need to redefine them because we don't want to depend from the Sysbox Makefile for the sysbox-in-docker Makefile. By breaking this dependency, we can now invoke the sysbox-in-docker make target from within the new sysbox-in-docker test container easily.

I see, makes sense.


> In general, Makefiles in sub-dirs should not depend on Makefiles in parent dirs.

Disagree here. This is a common pattern: attributes that affect the building environment (not the code itself) are usually place in outer Makefiles to avoid code-duplication. I think the exception is what we're doing here (code dup), but we have a good reason based on your explanation.

Reply:
Is fine to have the outer Makefile pass env variables to the inner Makefile, but it's not fine to render the inner Makefile unusable by itself (because is missing those env variables). That's what's being fixed.